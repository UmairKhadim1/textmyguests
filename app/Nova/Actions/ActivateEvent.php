<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Laravel\Nova\Actions\Action;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\ActionFields;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;

use App\Invoice;
use App\InvoiceLineItem;
use Carbon\Carbon;

class ActivateEvent extends Action
{
    use InteractsWithQueue, Queueable;

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        foreach ($models as $shindig) {
            if (!$shindig->payment_status) {
                $user = $shindig->owners()->first();
                // Create the invoice
                $invoice = Invoice::where('shindig_id', $shindig->id)->first();
                if ($invoice) {
                    $invoice->paid_amount = 0;
                    $invoice->paid_at = Carbon::now();
                    $invoice->save();
                } else {
                    $invoice = new Invoice();
                    $invoice->user_id = $user->id;
                    $invoice->shindig_id = $shindig->id;
                    $invoice->paid_amount = 0;
                    $invoice->paid_at = Carbon::now();
                    $invoice->save();
                }


                // Create the invoice line item
                $line_item = new InvoiceLineItem();
                $line_item->name = "Additionnal guests";
                $line_item->invoice_id = $invoice->id;
                $line_item->each_price = 0;
                $line_item->quantity = 500;
                $line_item->line_price = 0;
                $line_item->line_credits = 500;
                $line_item->save();

                $shindig->activatePaymentStatus();
            } else {
                return Action::danger('This event is already activated.');
            }
        }

        return Action::message('The event was successfully activated.');
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [];
    }
}
