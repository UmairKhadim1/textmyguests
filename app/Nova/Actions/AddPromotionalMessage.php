<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Laravel\Nova\Actions\Action;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\ActionFields;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;
use Laravel\Nova\Fields\Boolean;

use App\Shindig;
use App\Invoice;
use App\InvoiceLineItem;
use Carbon\Carbon;

class AddPromotionalMessage extends Action
{
    use InteractsWithQueue, Queueable;

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        foreach ($models as $shindig) {
            // Make sure the event is activated
            if (!$shindig->payment_status) {
                return Action::danger('You cannot add a promotional message to an event that is not activated yet.');
            }

            // Make sure the event does not already have a promotional message
            if ($shindig->messages()->where('promotion', 'share50')->exists()) {
                return Action::danger('This event already has a promotional message.');
            }

            // If we want to refund money (50$) to the shindig's owner who activated the event
            if ($fields->refund_money) {
                $invoice = $this->getInvoiceForRefund($shindig);

                if (!$invoice) {
                    return Action::danger('There is no invoice eligible for a refund.');
                }

                // Refund the price of the promotion
                try {
                    $this->refund($invoice, 50);
                } catch (\Exception $e) {

                    return Action::danger(json_encode($e));
                    return Action::danger("Sorry, an error occured while trying to refund invoice $invoice->id");
                }
            }

            // Add the promotional message
            $shindig->addPromotionalMessage();
        }
    }

    public function getInvoiceForRefund(Shindig $shindig)
    {
        // Get all invoices and make sure the user has a balance
        // greater than or equal to the refunded amount
        $amount = 50 * 100; // From $ to cents.
        $balance = 0;
        $invoiceToRefund = null;
        foreach ($shindig->invoices as $invoice) {
            $balance += $invoice->paid_amount;
            if ($invoice->paid_amount > $amount) {
                $invoiceToRefund = $invoice;
            }
        };

        if ($balance < $amount || !$invoiceToRefund) {
            return null;
        }

        return $invoiceToRefund;
    }

    public function refund(Invoice $invoiceToRefund, $amount)
    {
        // Configure stripe
        $stripeKey = env('STRIPE_SECRET');
        \Stripe\Stripe::setApiKey($stripeKey);

        $refund = \Stripe\Refund::create([
            'charge' => $invoiceToRefund->stripe_id,
            'amount' => $amount * 100, // in cents
        ]);

        // Create the invoice for the refund
        $invoice = new Invoice();
        $invoice->user_id = $invoiceToRefund->user_id;
        $invoice->shindig_id = $invoiceToRefund->shindig_id;
        $invoice->stripe_id = $refund->id;
        $invoice->paid_amount = $refund->amount * -1;
        $invoice->paid_at = Carbon::now();
        $invoice->card_type = $invoiceToRefund->card_type;
        $invoice->card_last4 = $invoiceToRefund->card_last4;
        $invoice->save();

        // Create the invoice line item
        $lineItem = new InvoiceLineItem();
        $lineItem->name = "Promotional message discount";
        $lineItem->invoice_id = $invoice->id;
        $lineItem->each_price = 0;
        $lineItem->quantity = 0;
        $lineItem->line_price = $refund->amount * -1;
        $lineItem->line_credits = 0;
        $lineItem->save();

        return $invoice;
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [
            Boolean::make('Refund money')
        ];
    }
}
