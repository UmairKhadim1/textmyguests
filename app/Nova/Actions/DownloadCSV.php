<?php

namespace App\Nova\Actions;


use Maatwebsite\LaravelNovaExcel\Actions\DownloadExcel;
use Maatwebsite\Excel\Concerns\WithMapping;
use PhpOffice\PhpSpreadsheet\Shared\Date;
use App\Shindig;

class DownloadCSV extends DownloadExcel implements WithMapping
{
    public $name = "Download CSV";

    /**
     * @param Shindig $shindig
     *
     * @return array
     */
    public function map($shindig): array
    {
        return [
            $shindig->id,
            $shindig->name,
            $shindig->event_date->toDateString(),
            $this->getEventOwners($shindig),
            $shindig->paidGuestLimit(),
            $shindig->guestCount(),
            $shindig->messageCount(),
        ];
    }

    public function getEventOwners(Shindig $shindig)
    {
        $owners = '';

        foreach ($shindig->owners as $key => $owner) {
            $owners .= $owner->email;
            if ($key !== count($shindig->owners) - 1) {
                $owners .= ', ';
            }
        }

        return $owners;
    }
 
}
