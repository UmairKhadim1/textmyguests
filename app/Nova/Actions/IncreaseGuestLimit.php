<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Laravel\Nova\Actions\Action;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\ActionFields;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;
use Laravel\Nova\Fields\Number;

use App\Invoice;
use App\InvoiceLineItem;
use Carbon\Carbon;

class IncreaseGuestLimit extends Action
{
    use InteractsWithQueue, Queueable;

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        foreach ($models as $shindig) {
            $user = $shindig->owners()->first();
            // Create the invoice
            $invoice = new Invoice();
            $invoice->user_id = $user->id;
            $invoice->shindig_id = $shindig->id;
            $invoice->paid_amount = 0;
            $invoice->paid_at = Carbon::now();
            $invoice->save();

            // Create the invoice line item
            $line_item = new InvoiceLineItem();
            $line_item->name = "Additionnal guests";
            $line_item->invoice_id = $invoice->id;
            $line_item->each_price = 0;
            $line_item->quantity = $fields->additionnal_guests;
            $line_item->line_price = 0;
            $line_item->line_credits = $fields->additionnal_guests;
            $line_item->save();

            if (!$shindig->payment_status && $fields->additionnal_guests > 0) {
                $shindig->activate();
            }
        }
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [
            Number::make('Additionnal guests')
        ];
    }
}
