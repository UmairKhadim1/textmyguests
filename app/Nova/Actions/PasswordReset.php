<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Laravel\Nova\Actions\Action;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\ActionFields;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Support\Facades\Password;

class PasswordReset extends Action
{
    use InteractsWithQueue, Queueable;

    public $name = "Trigger Password Reset";

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        foreach ($models as $model) {
            $credentials = ['email' => $model->email];
            $response = Password::sendResetLink($credentials);

            switch ($response) {
                case Password::RESET_LINK_SENT:
                    return Action::message('Password reset email was sent!');
                case Password::INVALID_USER:
                    return Action::danger('Something went wrong and the email was not sent.');
            }
        }
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [];
    }
}
