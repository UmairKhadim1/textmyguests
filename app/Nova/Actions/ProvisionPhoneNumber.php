<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Laravel\Nova\Actions\Action;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\ActionFields;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;

class ProvisionPhoneNumber extends Action
{
    use InteractsWithQueue, Queueable;

    public $confirmText = "Are you sure you want to provision a new phone number for this event?";
    public $confirmButtonText = "Provision";

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        foreach ($models as $shindig) {
            $shindig->provisionNewPhoneNumber();
        }

        return Action::message('A new phone number was successfully provisioned.');
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [];
    }
}
