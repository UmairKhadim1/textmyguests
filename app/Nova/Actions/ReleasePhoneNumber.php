<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\ActionFields;
use Illuminate\Queue\SerializesModels;
use Illuminate\Queue\InteractsWithQueue;
use Laravel\Nova\Actions\DestructiveAction;
use Illuminate\Contracts\Queue\ShouldQueue;
use Laravel\Nova\Actions\Action;

class ReleasePhoneNumber extends DestructiveAction
{
    use InteractsWithQueue, Queueable, SerializesModels;

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        foreach ($models as $phoneNumber) {

            if ($phoneNumber->dont_release) {
                return Action::danger('Cannot release this phone number.');
            } else {
                $phoneNumber->release();
                return Action::message('Phone number released.');
            }
        }
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [];
    }
}
