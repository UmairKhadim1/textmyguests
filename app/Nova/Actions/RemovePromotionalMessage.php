<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Laravel\Nova\Actions\Action;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\ActionFields;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;

class RemovePromotionalMessage extends Action
{
    use InteractsWithQueue, Queueable;

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        foreach ($models as $shindig) {
            $promotional_message = $shindig->messages()->where('promotion', 'share50')->first();

            // Make sure the event already have a promotional message
            if (!$promotional_message) {
                return Action::danger('This event does not have a promotional message.');
            }

            $promotional_message->delete();
            return Action::message('The promotional message was successfully removed.');
        }
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [];
    }
}
