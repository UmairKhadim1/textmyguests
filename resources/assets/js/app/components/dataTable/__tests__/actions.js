import React from 'react';
import { shallow } from 'enzyme';
import Actions from '../actions';

describe('Table Actions Tests', () => {
  it('should render without scratch', () => {
    const wrapper = shallow(<Actions />);
    expect(wrapper).toMatchSnapshot();
  });
});
