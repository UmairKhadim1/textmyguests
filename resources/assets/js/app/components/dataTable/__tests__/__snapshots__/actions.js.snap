// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Actions Tests should render without scratch 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Actions
    onDelete={[Function]}
    onEdit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <li>
          <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
          />
        </li>,
        <li>
          <Popconfirm
            cancelText="No"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            okText="Yes"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure you want to delete it"
            transitionName="zoom-big"
            trigger="click"
          >
            <Button
              block={false}
              ghost={false}
              icon="delete"
              loading={false}
              prefixCls="ant-btn"
              type="danger"
            />
          </Popconfirm>
        </li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            block={false}
            ghost={false}
            icon="edit"
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "icon": "edit",
            "loading": false,
            "onClick": [Function],
            "prefixCls": "ant-btn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Popconfirm
            cancelText="No"
            icon={
              <Icon
                theme="filled"
                type="exclamation-circle"
              />
            }
            okText="Yes"
            okType="primary"
            onConfirm={[Function]}
            placement="top"
            prefixCls="ant-popover"
            title="Are you sure you want to delete it"
            transitionName="zoom-big"
            trigger="click"
          >
            <Button
              block={false}
              ghost={false}
              icon="delete"
              loading={false}
              prefixCls="ant-btn"
              type="danger"
            />
          </Popconfirm>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cancelText": "No",
            "children": <Button
              block={false}
              ghost={false}
              icon="delete"
              loading={false}
              prefixCls="ant-btn"
              type="danger"
            />,
            "icon": <Icon
              theme="filled"
              type="exclamation-circle"
            />,
            "okText": "Yes",
            "okType": "primary",
            "onConfirm": [Function],
            "placement": "top",
            "prefixCls": "ant-popover",
            "title": "Are you sure you want to delete it",
            "transitionName": "zoom-big",
            "trigger": "click",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "ghost": false,
              "icon": "delete",
              "loading": false,
              "prefixCls": "ant-btn",
              "type": "danger",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <li>
            <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
            />
          </li>,
          <li>
            <Popconfirm
              cancelText="No"
              icon={
                <Icon
                  theme="filled"
                  type="exclamation-circle"
                />
              }
              okText="Yes"
              okType="primary"
              onConfirm={[Function]}
              placement="top"
              prefixCls="ant-popover"
              title="Are you sure you want to delete it"
              transitionName="zoom-big"
              trigger="click"
            >
              <Button
                block={false}
                ghost={false}
                icon="delete"
                loading={false}
                prefixCls="ant-btn"
                type="danger"
              />
            </Popconfirm>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              block={false}
              ghost={false}
              icon="edit"
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "ghost": false,
              "icon": "edit",
              "loading": false,
              "onClick": [Function],
              "prefixCls": "ant-btn",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Popconfirm
              cancelText="No"
              icon={
                <Icon
                  theme="filled"
                  type="exclamation-circle"
                />
              }
              okText="Yes"
              okType="primary"
              onConfirm={[Function]}
              placement="top"
              prefixCls="ant-popover"
              title="Are you sure you want to delete it"
              transitionName="zoom-big"
              trigger="click"
            >
              <Button
                block={false}
                ghost={false}
                icon="delete"
                loading={false}
                prefixCls="ant-btn"
                type="danger"
              />
            </Popconfirm>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cancelText": "No",
              "children": <Button
                block={false}
                ghost={false}
                icon="delete"
                loading={false}
                prefixCls="ant-btn"
                type="danger"
              />,
              "icon": <Icon
                theme="filled"
                type="exclamation-circle"
              />,
              "okText": "Yes",
              "okType": "primary",
              "onConfirm": [Function],
              "placement": "top",
              "prefixCls": "ant-popover",
              "title": "Are you sure you want to delete it",
              "transitionName": "zoom-big",
              "trigger": "click",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "block": false,
                "ghost": false,
                "icon": "delete",
                "loading": false,
                "prefixCls": "ant-btn",
                "type": "danger",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
